src/utils/currency.ts

```diff
@@ -185,3 +224,107 @@ export function tokenSymbolToCurrency(tokenSymbol: InterbtcPrimitivesTokenSymbol
     }
     throw new Error(`No entry provided for token symbol of type '${tokenSymbol?.type}'`);
 }
+
+/**
+ * Get foreign asset by its id.
+ * @param id The id of the foreign asset.
+ * @returns The foreign asset.
+ */
+export async function getForeignAssetFromId(api: ApiPromise, id: number | u32): Promise<ForeignAsset> {
+    const u32Id = id instanceof u32 ? id : newForeignAssetId(api, id);
+    const optionMetadata = await api.query.assetRegistry.metadata(u32Id);
+
+    if (!optionMetadata.isSome) {
+        return Promise.reject(new Error("Foreign asset not found"));
+    }
+    const currencyPart = DefaultAssetRegistryAPI.metadataToCurrency(optionMetadata.unwrap());
+    const coingeckoId = decodeBytesAsString(optionMetadata.unwrap().additional.coingeckoId);
+
+    const numberId = id instanceof u32 ? id.toNumber() : id;
+
+    return {
+        foreignAsset: {
+            id: numberId,
+            coingeckoId,
+        },
+        ...currencyPart,
+    };
+}
+
+/**
+ * Get underlying currency of lend token id,
+ *
+ * @param lendTokenId Currency id of the lend token to get currency from
+ * @returns Underlying CurrencyExt for provided lend token
+ */
+export async function getUnderlyingCurrencyFromLendTokenId(
+    api: ApiPromise,
+    lendTokenId: InterbtcPrimitivesCurrencyId
+): Promise<CurrencyExt> {
+    const underlyingCurrencyId = await api.query.loans.underlyingAssetId(lendTokenId);
+
+    const underlyingCurrency = await currencyIdToMonetaryCurrency(api, underlyingCurrencyId.unwrap());
```

Comment:
Do we need an `.isSome` check before unwrapping, or is it guaranteed to be set?

Reply:
Yes, this is always set if the lendToken exists. If not the function will throw - which is correct behaviour for invalid `lendTokenId` provided.